host/class/uvc/usb_host_uvc/private_include/uvc_types_priv.h

```diff
@@ -45,6 +45,7 @@ struct uvc_host_stream_s {
         uint8_t  bEndpointAddress;            // Streaming endpoint address. Needed for BULK Stream stop
 
         // USB host related members
+        uint8_t dev_addr;                     // USB device address
         usb_device_handle_t dev_hdl;          // USB device handle
```

Comment:
I'd like the UVC driver to follow USB Host Lib design principles.

USB Host Lib abstracts away the 'device address' and offers 'device handle'. Device handle contains information about device address, so we should not save both the handle and the address.

If you need to get the address from handle, you can call `usb_host_device_info()`

Reply:
you are right